rm/src/main/java/io/seata/rm/AbstractResourceManager.java

```diff
@@ -64,6 +75,8 @@ public Long branchRegister(BranchType branchType, String resourceId, String clie
             request.setBranchType(branchType);
             request.setApplicationData(applicationData);
 
+            StringUtils.checkDataSize(applicationData, "applicationData", appDataWarnPercent, appDataErrSize);
```

Comment:
There should be a switch whether to open the check, so as to avoid users to upgrade, in the runtime suddenly reported errors, the first version of the default value of false, and adjust to true in later versions

Reply:
And the 78 line of code should be better placed at 70 lines.